Explorer/Assets/Scripts/SceneRuntime/Apis/Modules/CommunicationsControllerApi/CommunicationsControllerAPIWrapper.cs

```diff
@@ -74,43 +83,118 @@ private void SendBinaryToParticipants(IList<object> dataList, string? recipient)
+            }
 
-            if (peerData != null)
-                for (var i = 0; i < peerData.Count; i++)
+            if (peerData.GetHashCode() != 0)
+            {
+                int peerDataCount;
+
+                using (var value = V8Value.New())
+                {
+                    peerData.GetNamedProperty("length", value);
+                    var decoded = value.Decode();
+                    peerDataCount = (int)decoded.GetNumber();
+                }
+
+                for (var i = 0; i < peerDataCount; i++)
                 {
-                    object? obj = peerData[i];
+                    V8Object perRecipientStruct;
 
-                    if (obj is IScriptObject perRecipientStruct)
+                    using (var value = V8Value.New())
                     {
-                        var recipient = (IList<object>)perRecipientStruct.GetProperty("address")!;
-                        var data = (IList<object>)perRecipientStruct.GetProperty("data")!;
+                        peerData.GetIndexedProperty(i, value);
+                        var decoded = value.Decode();
+                        perRecipientStruct = decoded.GetV8Object();
+                    }
 
-                        if (data.Count is 0)
-                            continue;
+                    V8Object data;
 
-                        if (recipient.Count is 0)
-                            SendBinaryToParticipants(data, null);
+                    using (var value = V8Value.New())
+                    {
+                        perRecipientStruct.GetNamedProperty("data", value);
+                        var decoded = value.Decode();
+                        data = decoded.GetV8Object();
+                    }
 
-                        foreach (object? address in recipient)
-                            if (address != null)
-                            {
-                                var stringAddress = (string)address;
+                    int dataCount;
 
-                                if (!string.IsNullOrEmpty(stringAddress))
-                                    SendBinaryToParticipants(data, stringAddress);
-                            }
+                    using (var value = V8Value.New())
+                    {
+                        data.GetNamedProperty("length", value);
+                        var decoded = value.Decode();
+                        dataCount = (int)decoded.GetNumber();
+                    }
+
+                    if (dataCount == 0)
+                        continue;
+
+                    V8Object recipient;
+
+                    using (var value = V8Value.New())
+                    {
+                        perRecipientStruct.GetNamedProperty("address", value);
+                        var decoded = value.Decode();
+                        recipient = decoded.GetV8Object();
                     }
+
+                    int recipientCount;
+
+                    using (var value = V8Value.New())
```

Comment:
We have quite a bit of repetition of this using block, could we wrap this to make this function cleaner?

Reply:
We could.